### DISCUSSION TITLE
Bug (or feature??) in function "tud_hid_n_report"

### DISCUSSION THREAD

**Discussion by Commotari** [hathach/tinyusb#2078] [General]
in function tud_hid_n_report the length is getting limited according the bufferlength "CFG_TUD_HID_EP_BUFSIZE" that is configured in tusb_config.h.
if using the function with bigger package lengths of course it does not work. but also it does not tell that anything went wrong.
it just sends the package out with a limited length and returns success.
on my windows application that package never arrives because Windows somehow gets very mad when a package comes along that does not fit the USB descriptor. not only does it throw away that package but it also disconnects the usb connection somehow.

i would suggest a fix:
if the length does not fix the buffer size, then the function "tud_hid_n_report" should not send a cripled package but rather should return false, so that the application can see that something went wrong with sending the package.

beside that, thanks to all developers, you are doing a fantastic job!

**Comment by hathach** [comment]
you're right, we should refuse to make the transfer and return with error. I created an issue #2087 based on this to reminde myself in the future. If possible, please consider to make an PR for this
